### DISCUSSION TITLE
Feature request:  Variables in fields

### DISCUSSION THREAD

**Discussion by txjmb** [grafana/grafana-json-datasource#249] [Ideas]
Thank you for the excellent plugin!

We have a feature request to enable using variables in field definition jsonpaths.  This will require some sort of escaping I'm sure because $ is important in jsonpath, but I'm sure there's some way we can come up with to reasonably escape variables in the jsonpath.  This would allow parts or all of the jsonpath to be driven by variables enabling dropdown-driven changes to that part of the query when necessary.

Happy to pitch in and help if this feature makes sense for everyone.

Thank you for your consideration.

**Comment by marcusolsson** [comment]
I can definitely see how this would be useful. I don't actually know whether $ is used for anything other than the root, and whether we could assume it's always followed by a period. I'd be happy to accept this if we can find a way to avoid conflicts.

**Comment by txjmb** [comment]
Ok, thank you.  We will do a fork and a PR as soon as we are able.  After reviewing jsonpath documentation, it does seem that $ is solely used to refer to the root of the json document.  I can't think of ways it could be used outside of the first character.  At least not yet...

**Comment by hafeja** [comment]
Hello @txjmb ,

it looks like this feature request has the same scope like https://github.com/marcusolsson/grafana-json-datasource/issues/95.

It would be great if you find a solution for it.

**Comment by txjmb** [comment]
Hmm...  Thanks for pointing this out @hafeja , this does seem to be related, although the use of the data source is in a different context (setting variables).  I'll take a look...
